components/ai_chat/resources/untrusted_conversation_frame/components/assistant_task/use_extract_task_data.test.ts

```diff
@@ -0,0 +1,345 @@
+    it('should extract an important tool', () => {
+      const assistantEntries: Mojom.ConversationTurn[] = [
+        createConversationTurnWithDefaults({
+          characterType: Mojom.CharacterType.ASSISTANT,
+          events: [
+            getCompletionEvent('Navigating...'),
+            getToolUseEvent({
+              toolName: Mojom.NAVIGATE_TOOL_NAME,
+              id: '1',
+              argumentsJson: '{"url": "https://example.com"}',
+              output: undefined,
+            }),
+          ],
+        }),
+        createConversationTurnWithDefaults({
+          characterType: Mojom.CharacterType.ASSISTANT,
+          events: [
+            getCompletionEvent('Something else'),
+            getToolUseEvent({
+              toolName: 'other-tool',
+              id: '1',
+              argumentsJson: '{}',
+              output: undefined,
+            }),
+          ],
+        }),
+      ]
+
+      const { result } = renderHook(() => useExtractTaskData(assistantEntries))
+
+      expect(result.current.importantToolUseEvents).toHaveLength(1)
+      expect(result.current.importantToolUseEvents[0].toolName).toBe(
+        Mojom.NAVIGATE_TOOL_NAME,
+      )
+    })
+
+    it('should only keep the most recent important tool of each type', () => {
+      const assistantEntries: Mojom.ConversationTurn[] = [
+        createConversationTurnWithDefaults({
+          characterType: Mojom.CharacterType.ASSISTANT,
+          events: [
+            getCompletionEvent('First navigation'),
+            getToolUseEvent({
+              toolName: Mojom.NAVIGATE_TOOL_NAME,
+              id: '1',
+              argumentsJson: '{"url": "https://first.com"}',
+              output: undefined,
+            }),
+          ],
+        }),
+        createConversationTurnWithDefaults({
+          characterType: Mojom.CharacterType.ASSISTANT,
+          events: [
+            getToolUseEvent({
+              toolName: Mojom.NAVIGATE_TOOL_NAME,
+              id: '2',
+              argumentsJson: '{"url": "https://second.com"}',
+              output: undefined,
+            }),
+            getCompletionEvent('Second navigation'),
+          ],
+        }),
+        createConversationTurnWithDefaults({
+          characterType: Mojom.CharacterType.ASSISTANT,
+          events: [
+            getToolUseEvent({
+              toolName: Mojom.NAVIGATE_TOOL_NAME,
+              id: '3',
+              argumentsJson: '{"url": "https://third.com"}',
+              output: undefined,
+            }),
+            getCompletionEvent('Third navigation'),
+          ],
+        }),
+      ]
+
+      const { result } = renderHook(() => useExtractTaskData(assistantEntries))
+
+      // Should only have 1 navigate tool event (the most recent one)
+      expect(result.current.importantToolUseEvents).toHaveLength(1)
+      expect(result.current.importantToolUseEvents[0].id).toBe('3')
+    })
+
+    it('should not include non-important tools', () => {
```

Comment:
can we have a test mixing important/non-important

Reply:
We do have that already with `'should extract an important tool'` or were you thinking to have some different kind of mix?